pkg/utils/kubernetes/status.go

```diff
@@ -195,7 +195,10 @@ func SetAcceptedConditionOnGateway(resource ConditionsAndListenerConditionsAndGe
 	}
 }
 
-func areAllConditionsHaveTrueStatus(resource ConditionsAware) bool {
+// AreAllConditionsHaveTrueStatus checks if all the conditions on the resource are in the True state.
+// It skips the Programmed condition as that particular condition will be set based on
+// the return value of this function.
+func AreAllConditionsHaveTrueStatus(resource ConditionsAware) bool {
 	for _, condition := range resource.GetConditions() {
 		if condition.Type == string(gatewayv1.GatewayConditionProgrammed) {
```

Comment:
The logic is not so intuitive here. I think the criteria to return false should be "Type is not `Programmed` or `Ready` and Status is `False`". So it looks clearer to move the `condition.Type != string(consts.ReadyType)` to
 
```
// Skip check if the condition type is "Programmed" or "Ready"
if condition.Type == string(gatewayv1.GatewayConditionProgrammed) || condition.Type == string(consts.ReadyType) {
    continue
} 
```

Reply:
Addressed in 00f6973af3a0f6b8f8f6f443609b79d34159432b